pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -276,7 +276,7 @@ func (og *operationGenerator) GenerateDetachVolumeFunc(
 			err = og.verifyVolumeIsSafeToDetach(volumeToDetach)
 		}
 		if err == nil {
-			err = volumeDetacher.Detach(volumeName, volumeToDetach.NodeName)
+			err = volumeDetacher.Detach(volumeToDetach.VolumeSpec, volumeName, volumeToDetach.NodeName)
```

Comment:
Oh, I see, you're using it for your new local (cloud-free) cinder implementation. The cloud provider implementations get credentials through external mechanism. For the local implementation you rely on reading credentials from a secret. And in order to get `SecretRef *LocalObjectReference`, which is available in the spec, and the pod namespace you need the volume spec on the detach.

```GO
func (cdl *cdManagerLocal) Detach(spec *volume.Spec, deviceName string, nodeName types.NodeName) error {
```

The problem with this, as stated above, is that the full `volumeSpec` passed into `volumeToDetach` can't really be relied on. If the controller binary is restarted between attach and detach and the PV or pod w/the volume def is deleted in the meantime, then the spec information is lost. Meaning if this happens (or if you delete your secret before detach), your volume gets stuck attached and you need to go and manually detach it. Would you be ok with this scenario?